crates/volta-core/src/tool/registry.rs

```diff
@@ -30,11 +30,13 @@ cfg_if! {
     }
 }
 
-pub fn public_registry_package(package: &str, version: &str) -> String {
+// need package and filename for namespaced tools like @yarnpkg/cli-dist, which is located at
+//   https://registry.npmjs.org/@yarnpkg/cli-dist/-/cli-dist-1.2.3.tgz
+pub fn public_registry_package(package: &str, filename: &str, version: &str) -> String {
```

Comment:
Would it possibly make sense to split this into two functions? Something like the existing `public_registry_package` and then `scoped_public_registry_package` That does this:

```rust
pub fn scoped_public_registry_package(scope: &str, package: &str, version: &str) -> String {
    format!(
        "{}/{}/-/{}-{}.tgz",
        public_registry_index(scope),
        package,
        package,
        version
    )
}
```

Reply:
Totally makes sense. I split that up like you suggested in db9e6d1d0e1a2ca910231af66ce65b8f9416f382